src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,67 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int FIGURE_RANDOM_PARAMETR = 20;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private String colorFigure = colorSupplier.getRandomColor();
+
+    public Figure getRandomFigure() {
+        int randomNumber = random.nextInt(FIGURE_COUNT);
+
+        switch (randomNumber) {
+            case 0:
+                return createCircle();
+            case 1:
+                return createSquare();
+            case 2:
+                return createRightTriangle();
+            case 3:
+                return createRectangle();
+            case 4:
+                return createIsoscelesTrapezoid();
+            default:
+                return new Circle(10, "white");
+
+        }
+    }
+
+    private Circle createCircle() {
+        int radius = FIGURE_RANDOM_PARAMETR;
+        return new Circle(radius,colorFigure);
+
+    }
+
+    private Square createSquare() {
+        int side = FIGURE_RANDOM_PARAMETR;
+        return new Square(side, colorFigure);
+    }
+
+    private RightTriangle createRightTriangle() {
+        int firstLeg = FIGURE_RANDOM_PARAMETR;
+        int secondLeg = FIGURE_RANDOM_PARAMETR;
+        return new RightTriangle(firstLeg, secondLeg, colorFigure);
+    }
+
+    private Rectangle createRectangle() {
+        int firstSide = FIGURE_RANDOM_PARAMETR;
+        int secondSide = FIGURE_RANDOM_PARAMETR;
+        return new Rectangle(firstSide, secondSide, colorFigure);
+    }
+
+    private IsoscelesTrapezoid createIsoscelesTrapezoid() {
+        int firstDistance = FIGURE_RANDOM_PARAMETR;
+        int secondDistance = FIGURE_RANDOM_PARAMETR;
+        int height = FIGURE_RANDOM_PARAMETR;
+        return new IsoscelesTrapezoid(firstDistance, secondDistance, height, colorFigure);
```

Comment:
too many duplicates, please think of better approach. 
Also FIGURE_RANDOM_PARAMETR; is 20, so each figure will have 20 as parameter, there's no random

Reply:
Not fixed.
I suggest create method and use it directly in constructor for each type of shape
```
public int randomSize() {
        return random.nextInt() * FIGURE_RANDOM_PARAMETR;
}
```